### PR TITLE
[AKS] Be able to create AKS with virtual node enabled

### PR DESCRIPTION
**Pull request by gossion**
**Description**  

Be able to create with virtual node enabled. 

**Testing Guide**  

**History Notes**  

[AKS] support virtual-node with aks create: `az aks create --enable-addons virtual-node`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Comment by gossion** [comment]
@jluk @feiskyer I am preparing a PR to support `--enable-addon virtual-node` for `az aks create`, can you take a look before I publish the PR? 
Note: it requires `"--aci-subnet-name"(new param) and "--vnet-subnet-id"(existing param)`

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -2459,6 +2462,15 @@ def _handle_addons_args(cmd, addons_str, subscription_id, resource_group_name, a
     if 'azure-policy' in addons:
         addon_profiles['azurepolicy'] = ManagedClusterAddonProfile(enabled=True)
         addons.remove('azure-policy')
+    if 'virtual-node' in addons:
+        if not aci_subnet_name or not vnet_subnet_id:
+            raise CLIError('"--enable-addons virtual-node" requires "--aci-subnet-name" and "--vnet-subnet-id".')
+        # TODO: how about aciConnectorwindows, what is its addon name?
+        addon_profiles['aciConnectorLinux'] = ManagedClusterAddonProfile(
```
Comment (MikkelHegn) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:2469]: I think it's the 'aciConnector' addon profile, not 'aciConnectorLinux' we need to use. See: https://github.com/Azure/azure-cli/blob/0c88076fafb279665529bfa8178e76eab7c2bbfe/src/azure-cli/azure/cli/command_modules/acs/custom.py#L1999 and the implementation in the extension CLI: https://github.com/Azure/azure-cli-extensions/blob/3c7edde5e5475a9c1f2a8e9b602923da432d6f93/src/aks-preview/azext_aks_preview/custom.py#L2513

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 2]
```diff
@@ -2459,6 +2462,15 @@ def _handle_addons_args(cmd, addons_str, subscription_id, resource_group_name, a
     if 'azure-policy' in addons:
         addon_profiles['azurepolicy'] = ManagedClusterAddonProfile(enabled=True)
         addons.remove('azure-policy')
+    if 'virtual-node' in addons:
+        if not aci_subnet_name or not vnet_subnet_id:
+            raise CLIError('"--enable-addons virtual-node" requires "--aci-subnet-name" and "--vnet-subnet-id".')
+        # TODO: how about aciConnectorwindows, what is its addon name?
```
Comment (MikkelHegn) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:2468]: We do not support Windows today.

File: src/azure-cli/azure/cli/command_modules/acs/custom.py (hunk 1 above)
Comment (gossion) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:2469]: Thanks @MikkelHegn, it is the profile should be `aciConnectorLinux` for Linux, see `addon += os_type` in https://github.com/Azure/azure-cli/blob/0c88076fafb279665529bfa8178e76eab7c2bbfe/src/azure-cli/azure/cli/command_modules/acs/custom.py#L2369-L2371, which is currently defined in aks-rp. 
Currently It is name is only `aciConnector`, which would be used for both Linux and Windows, and that is why I am asking what is its name for Windows.

Comment (feiskyer) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:2469]: Windows is not supported yet, but I think `addon += os_type` should also work when we add Windows support in the future.

Comment (gossion) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:2469]: yes, `aciConnector`  equals to `aciConnectorLinux`  today, and we are enabling it with param `--enable-addon virtual-node`.
For further view, what if we enable it for Windows in the future? should it be `--enable-addon virtual-node-windows` ?

**Comment by yungezz** [comment]
hi @gossion are you expecting the PR to catch Ignite release? we'll cut build tomorrow.

**Comment by gossion** [comment]
Thanks for reminder @yungezz , this is not urgent. you can go ahead without this PR.

**Review by arrownj — APPROVED** [review]
The code looks good to me.

**Comment by arrownj** [comment]
We will merge this PR after the ignite build if no more concerns addressed.

**Review by feiskyer — APPROVED** [review]
/lgtm

**Pull request merged**